go-controller/pkg/crd/userdefinednetwork/v1/shared.go

```diff
@@ -119,6 +130,43 @@ type Layer2Config struct {
 	// +optional
 	Subnets DualStackCIDRs `json:"subnets,omitempty"`
 
+	// reservedSubnets specifies a list of CIDRs reserved for static IP assignment, excluded from automatic allocation.
+	// reservedSubnets is optional. When omitted, all IP addresses in `subnets` are available for automatic assignment.
+	// IPs from these ranges can still be requested through static IP assignment in pod annotations.
+	// Each item should be in range of the specified CIDR(s) in `subnets`.
+	// The maximum number of entries allowed is 25.
+	// The format should match standard CIDR notation (for example, "10.128.0.0/16").
+	// This field must be omitted if `subnets` is unset or `ipam.mode` is `Disabled`.
+	// +optional
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=25
+	ReservedSubnets []CIDR `json:"reservedSubnets,omitempty"`
```

Comment:
Is it difference then the localnet exlcudeSubentes?
If its possible I would call it excludeSubnet for consistent naming across typologies.

Reply:
> It seems that reservedSubnetes are treated the same as with localnets excludeSubnets, right?

That's  not the case, while excludeSubnets are completely removed from the allocation pool the user can still use addresses from the `reservedSubnetes` by requesting a specific one.